crates/primitives/src/block.rs

```diff
@@ -83,6 +100,38 @@ impl Deref for SealedBlock {
     }
 }
 
+/// Sealed block with senders recovered from transactions.
+#[derive(Debug, Clone, PartialEq, Eq, Default)]
+pub struct SealedBlockWithSenders {
+    /// Sealed block
+    pub block: SealedBlock,
+    /// List of senders that match trasanctions from block.
+    pub senders: Vec<Address>,
+}
```

Comment:
Why not a `SealedBlock` with `TransactionSignedEcRecovered`'s instead of `Transaction`s? I generally am not sure why the executor moved away from using those to using `Option<Vec<Address>>` everywhere

Reply:
It becomes harder to go from one type to another and to seal/unseal block, with separated senders it is easier to just add or not add senders.